Modify rule S6232: review LaYC format (#2614)
SonarSource/rspec: rules/S6232/cfamily/rule.adoc

> `union` is not suitable for type-punning in {cpp} code.
> Using `union` for type-punning leads to undefined behavior, it is unsafe,
> and it has well-defined safe alternatives that are just as fast.

Reviewer: ```suggestion
`union` is unsuitable for type-punning in {cpp} code, leading to undefined behavior. There are well-defined safe alternatives that are just as fast.
```
